frontend/src/components/editor/cell/PendingDeleteConfirmation.tsx

```diff
@@ -0,0 +1,193 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { useAtom, useAtomValue } from "jotai";
+import { AlertTriangleIcon } from "lucide-react";
+import React, { useEffect } from "react";
+import { formatElapsedTime } from "@/components/editor/cell/CellStatus";
+import { CellLink } from "@/components/editor/links/cell-link";
+import { Button } from "@/components/ui/button";
+import { useCellNames } from "@/core/cells/cells";
+import type { CellId } from "@/core/cells/ids";
+import { pendingDeleteCellsAtom } from "@/core/cells/pending-delete";
+import { variablesAtom } from "@/core/variables/state";
+import { cn } from "@/utils/cn";
+import { useDeleteCellCallback } from "./useDeleteCell";
+
+export const PendingDeleteInformation: React.FC<
+  PendingDeleteInformationProps
+> = ({ executionTimeMs, cellId }) => {
+  const pendingCells = useAtomValue(pendingDeleteCellsAtom);
+  if (!pendingCells.has(cellId)) {
+    return null;
+  }
+  return (
+    <PendingDeleteInformationInternal
+      cellId={cellId}
+      executionTimeMs={executionTimeMs}
+    />
+  );
+};
+
+interface PendingDeleteInformationProps {
+  cellId: CellId;
+  executionTimeMs: number;
+}
+
+const PendingDeleteInformationInternal: React.FC<
+  PendingDeleteInformationProps
+> = ({ cellId, executionTimeMs }) => {
+  const variables = useAtomValue(variablesAtom);
+  const cellNames = useCellNames();
+  const [pendingCells, setPendingCells] = useAtom(pendingDeleteCellsAtom);
+  const deleteCell = useDeleteCellCallback();
+
+  const downstream = new Set<CellId>();
+  Object.values(variables).forEach((v) => {
+    if (v.declaredBy.includes(cellId)) {
+      v.usedBy.forEach((id) => downstream.add(id));
+    }
+  });
+  const downstreamCells = [...downstream].map((id) => ({
+    cellId: id,
+    name: cellNames[id] || id,
+  }));
+
+  const hasExpensiveExecution = executionTimeMs > 2000;
+  const hasDependencies = downstreamCells.length > 0;
+  const isExpensiveOrHasDeps = hasExpensiveExecution || hasDependencies;
+  const isMultiPending = pendingCells.size > 1;
+
+  // Auto-delete if not expensive and not multi-pending
+  useEffect(() => {
+    if (!isMultiPending && !isExpensiveOrHasDeps) {
```

Comment:
maybe pull this out `const autoDelete = !isMultiPending && !isExpensiveOrHasDeps` and reuse below